virtcontainers/kata_agent.go

```diff
@@ -1550,6 +1556,61 @@ func (k *kataAgent) handleEphemeralStorage(mounts []specs.Mount) []*grpc.Storage
 	return epheStorages
 }
 
+// handleHugePages handles hugepages storage by
+// creating a Storage from corresponding source of the mount point
+func (k *kataAgent) handleHugepages(mounts []specs.Mount, hugepageLimits []specs.LinuxHugepageLimit) []*grpc.Storage {
+	//Map to hold the total memory of each type of hugepages
+	optionsMap := make(map[string]string)
+	var HugePageSizeUnitList = []string{"B", "K", "M", "G", "T", "P"}
+
+	for _, hp := range hugepageLimits {
+		if hp.Limit != 0 {
+			k.Logger().WithField("Pagesize", hp.Pagesize).Info("hugepage request")
+			k.Logger().WithField("Limit", hp.Limit).Info("hugepage request")
+			//example Pagesize 2MB, 1GB etc. The Limit are in Bytes
+			pageSize, _ := units.RAMInBytes(hp.Pagesize)
+			pageSizeStr := units.CustomSize("%g%s", float64(pageSize), 1024.0, HugePageSizeUnitList)
```

Comment:
I don't understand this - why not use the `units.*` package to deal with the suffixes rather than specifying your own list...?

Reply:
I used the CustomSize method to explicitly indicate the acceptable unit options for hugepage size. I'll recheck if there is a different method without the need to specify the custom format.

Reply:
I don't really understand pageSizeStr at all.  You need to convert the (underspecified) string from the spec into a pagesize in bytes, but why do you then need a string.  You should be able to do your comparisons just based on the absolute size from then onwards.

Reply:
@dgibson regarding your comment on pageSizeStr, I'm unable to provide inline response. Not sure why. So putting it here.
The mount options uses something like pagesize=<pageSizeStr> eg. pagesize=1G. So using pageSizeStr to refer to the pagesize (1G, 2M etc.). Further, I use this to get the actual size from the map to create the final mount option string.

Reply:
The mount options allow "pagesize=1G" as syntactic sugar, but "pagesize=1073741824" will work too.  It makes sense to allow user input to use suffixes for brevity, but internally you should be able to use pagesize values in bytes everywhere.

Reply:
@dgibson addressed your comment, ptal